src/backend/utils/adt/jsonpath_exec.c

```diff
@@ -573,7 +573,7 @@ tsql_openjson_with_columnize(Jsonb *jb, char *col_info)
 		if (token)
 		{
 			token = strtok(NULL, ")");
-			if (token)
+			if (token && atoi(token) > 0)
```

Comment:
what was it returning for max?

Reply:
For nvarchar(max), token is -8000, here col_size was set as -8000 
Add this condition will make col_size the default size INT_MAX